setup/steps/confirm_test_command.go

```diff
@@ -0,0 +1,75 @@
+package steps
+
+import (
+	"fmt"
+
+	survey "github.com/AlecAivazis/survey/v2"
+	"github.com/guregu/null"
+	s "github.com/pganalyze/collector/setup/state"
+)
+
+var ConfirmTestCommand = &s.Step{
+	Description: "Run a test command in Postgres to confirm everything is working",
+	Check: func(state *s.SetupState) (bool, error) {
+		needsSleep := (state.Inputs.ConfirmSetUpLogInsights.Valid && state.Inputs.ConfirmSetUpLogInsights.Bool) ||
+			(state.Inputs.ConfirmSetUpAutomatedExplain.Valid && state.Inputs.ConfirmSetUpAutomatedExplain.Bool)
+		return !needsSleep || state.DidPgSleep ||
+			(state.Inputs.ConfirmRunTestCommand.Valid && !state.Inputs.ConfirmRunTestCommand.Bool), nil
+	},
+	Run: func(state *s.SetupState) error {
+		row, err := state.QueryRunner.QueryRow(
+			"SELECT coalesce(max(setting::float), 0) / 1000 * 1.2 from pg_settings where name IN ('log_min_duration_statement', 'auto_explain.log_min_duration')",
+		)
+		if err != nil {
+			return err
+		}
+		naptime := row.GetFloat(0)
+		var runTestCommand bool
+		if state.Inputs.Scripted {
+			if state.Inputs.ConfirmRunTestCommand.Valid {
+				runTestCommand = state.Inputs.ConfirmRunTestCommand.Bool
+			}
+		} else {
+			var testCmdType string
+			if naptime > 0 {
+				testCmdType = "pg_sleep"
+			} else {
+				testCmdType = "RAISE NOTICE"
+			}
+
+			hasAutomatedExplain := state.Inputs.ConfirmSetUpAutomatedExplain.Valid && state.Inputs.ConfirmSetUpAutomatedExplain.Bool
+			var features string
+			if hasAutomatedExplain {
+				features = "Log Insights and Automated EXPLAIN"
+			} else {
+				features = "Log Insights"
+			}
+			err := survey.AskOne(&survey.Confirm{
+				Message: fmt.Sprintf("Run a test command (%s) to confirm %s configuration?", testCmdType, features),
+				Default: true,
+				Help:    "You should see results in pganalyze a few seconds after the query completes",
```

Comment:
This here is a bit tricky, because in the initial setup, you haven't actually reloaded the collector yet (nor sent a test snapshot), so this doesn't actually work as intended.

If the customer goes fast enough, and runs the test command within 1 minute of the setup finishing, they would still get a benefit from this (since we go back 1 minute in the log tail), but not sure if we can rely on that.

It might be better if we move this logic to the test command in the main collector code instead? (especially the pg_sleep for verifying the EXPLAIN setup would be very useful there -- and I think we already issue the NOTICE, so I'm not sure if we need to do anything in that regard)

Reply:
We do reload the collector in an earlier step. Is that not enough without the test snapshot? You're right that consolidating this is probably a good idea, though. Should we remove this step entirely (though merging the sleep functionality into the collector test command as you suggest)?

Reply:
Ah, I missed that - that would be enough to make it show up, since we run this step after the reload.

Since we also have you run the test command though, maybe still makes sense to drop it here and only doing in the test command? (but I'm also good to leave as is)

Reply:
Ok, let's drop it then and move the sleep logic to the test command--I think that makes more sense. Should we drop the collector reload step then as well (and rely on the test command for that)?

Reply:
@uhoh-itsmaciek Good question, I'm actually not sure whether we want to keep the reload step in the setup itself.

The benefit of keeping the reload step is that in case someone doesn't run the test command, or forgets to run it with reload, it would still give them data in-app.

That actually makes me think again, maybe we should change the in-app experience to be an explicit "Check for new test snapshots" button (or similar), and actually run the `--test --reload` command for you at the end of the setup?

(our main motivation for not doing so previously was that we were worried the in-app experience would have timed out already - maybe we can avoid that problem & make it one step overall)